tests/Aspire.Hosting.Azure.Tests/AzureEventHubsExtensionsTests.cs

```diff
@@ -583,4 +583,62 @@ public void AzureEventHubsAppliesAzureFunctionsConfiguration()
         Assert.Equal("cg1", target["Aspire__Azure__Messaging__EventHubs__EventHubBufferedProducerClient__cg1__ConsumerGroup"]);
         Assert.Equal("hub1", target["Aspire__Azure__Messaging__EventHubs__EventHubBufferedProducerClient__cg1__EventHubName"]);
     }
+
+    [Fact]
+    [RequiresDocker]
+    public async Task AzureEventHubs_WithPersistentLifetime_ReusesContainers()
+    {
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(10));
+        using var builder1 = TestDistributedApplicationBuilder.Create(testOutputHelper);
+
+        var resource = builder1.AddAzureEventHubs("resource")
+                      .RunAsEmulator(x => x.WithLifetime(ContainerLifetime.Persistent));
+
+        resource.AddHub("hub");
+
+        using var app = builder1.Build();
+
+        await app.StartAsync(cts.Token);
+
+        var rns = app.Services.GetRequiredService<ResourceNotificationService>();
+
+        await rns.WaitForResourceHealthyAsync(resource.Resource.Name, cts.Token);
+
+        var resourceEvent = await rns.WaitForResourceAsync("resource", e => e.Snapshot.State?.Text == KnownResourceStates.Running, cts.Token);
+        var ehContainerId1 = resourceEvent.Snapshot.Properties.FirstOrDefault(x => x.Name == "container.id")?.Value?.ToString();
+
+        resourceEvent = await rns.WaitForResourceAsync("resource-storage", e => e.Snapshot.State?.Text == KnownResourceStates.Running, cts.Token);
+        var storageContainerId1 = resourceEvent.Snapshot.Properties.FirstOrDefault(x => x.Name == "container.id")?.Value?.ToString();
+
+        await app.StopAsync(cts.Token).WaitAsync(TimeSpan.FromMinutes(1));
+
+        using var builder2 = TestDistributedApplicationBuilder.Create(testOutputHelper);
```

Comment:
This test could use this new functionality:

https://github.com/dotnet/aspire/issues/4878#issuecomment-2648623757

That way we knew the app was stopped.

cc @davidfowl @karolz-ms

Reply:
It's on by default. But I decided to disable it on purpose, it's not required for the test (we check we don't have a different container, the state of the previous one doesn't matter) and not waiting makes the test faster (19s -> 12s)